src/defold/Buffer.hx

```diff
@@ -63,17 +63,18 @@ extern class Buffer {
         @param stream_name the stream name
         @return the data stream
     **/
-    @:pure static function get_stream(buffer:Buffer, stream_name:HashOrString):BufferStream;
+    @:pure static function get_stream(buffer:defold.types.Buffer, stream_name:HashOrString):BufferStream;
 }
 
 /**
     A buffer stream handle returned by `Buffer.get_stream`.
 **/
-extern class BufferStream {}
+extern class BufferStream extends lua.Table.AnyTable {}
 
 /**
     Structure for buffer stream declaration used in `Buffer.create`.
 **/
+@:pure
```

Comment:
What is this for? I've never seen `@:pure` used on a `typedef` before.

Reply:
Currently Buffer.create() doesn't work because of the metadata fields in the BufferStreamDeclaration Lua table. I thought @:pure could fix that. Unfortunately it doesn't.
```
var arr = new Array<BufferStreamDeclaration>();
arr.push({
	count: 1,
	name: "pixels",
	type: VALUE_TYPE_UINT8
});
var b = Buffer.create(1, Table.fromArray(arr));
```
Gives error
```
buffer.create: Unknown index name: __fields__
```